binding/dotnet/Cheetah/Cheetah.cs

```diff
@@ -363,5 +393,36 @@ private string[] GetMessageStack()
 
             return messageStack;
         }
+
+        /// <summary>
+        /// Retrieves a list of hardware devices that can be specified when constructing the model.
+        /// </summary>
+        /// <returns>An array of available hardware devices.</returns>
+        /// <exception cref="CheetahException">Thrown when an error occurs while retrieving the hardware devices.</exception>
+        public static string[] GetAvailableDevices()
+        {
+            IntPtr hardwareDevicesPtr;
+            int numDevices;
+            CheetahStatus status = pv_cheetah_list_hardware_devices(
+                out hardwareDevicesPtr,
+                out numDevices);
+            if (status != CheetahStatus.SUCCESS)
+            {
+                throw CheetahStatusToException(
+                    status,
+                    "Get available devices failed");
+            }
```

Comment:
you're missing the error stack here. Make `GetMessageStack()` static and then include the stack dump here